### PR TITLE
Feature - Strict Query Validation

### PR DESCRIPTION
**Pull request by kodumbeats**
This PR introduces a new Queries validator to validate an array of Query objects by:
- Individually validating each Query object
- **Strict validation**: validate the query params can all be found in one index
- Checks if index is queued for creation during strict validation

Test plan:
Unit tests included

TODO:

- [ ] Handle index order (ASC, DESC, etc)

### CODE REVIEW

**Comment by eldadfux** [comment]
@kodumbeats I might also add a new collection attributed called "indexesInQueue" to hold pending indexes that are still being created, this will allow us to give a better error messages in this validator.

**Comment by kodumbeats** [comment]
@eldadfux I'm not 💯 on what is best to pass to the Queries validator constructor. I landed on this approach in acefb92 because passing in `Utopia\Database\Database` made the code really messy, and I lost the story.

File: src/Database/Validator/Queries.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        /**
+         * @var bool
+         */
+        $queued = false;
+        
+        // Return false if attributes do not exactly match an index
+        if ($this->strict) {
+            // look for strict match among indexes
+            foreach ($this->indexes as $index) {
+                if ($index['attributes'] === $queries) {
```
Comment (eldadfux) [inline src/Database/Validator/Queries.php:103]: I'm not sure this array comparison in PHP actually work

File: src/Database/Validator/Queries.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    $indexId = $index['$id']; 
+                }
+            }
+
+            if (!$indexId) {
+                // check against the indexesInQueue
+                foreach ($this->indexesInQueue as $index) {
+                    if ($index['attributes'] === $queries) {
+                        $queued = true;
```
Comment (eldadfux) [inline src/Database/Validator/Queries.php:112]: If an index was found in the queue, we can stop the iteration by returning false. It will be faster.

**Review by eldadfux — CHANGES_REQUESTED** [review]
Looks very good, left some minor comments. When ready I can merge it and start playing around with it :)

File: src/Database/Validator/Queries.php (hunk 1 above)
Comment (kodumbeats) [reply src/Database/Validator/Queries.php]: Moved array comparison logic to its own method. 👍🏻

**Review by eldadfux — APPROVED** [review]

**Pull request merged**